packages/cli/src/lib/setup/setupInstall.ts

```diff
@@ -322,13 +351,13 @@ function Install(options) {
                 }
                 if (packetDirName.includes('/') && !packetDirName.startsWith('@')) {
                     // only scoped packages (e.g. @types/node ) may have a slash in their path
-                    packetDirName = packetDirName.substr(packetDirName.lastIndexOf('/') + 1);
+                    packetDirName = packetDirName.substring(packetDirName.lastIndexOf('/') + 1);
                 }
             }
             const installDir = tools.getAdapterDir(packetDirName);
 
             // inject the installedFrom information in io-package
-            if (fs.existsSync(installDir)) {
+            if (installDir !== null && fs.existsSync(installDir)) {
```

Comment:
```suggestion
            if (installDir && fs.existsSync(installDir)) {
```
I'd catch empty strings and undefined too.